libraries/Microsoft.Bot.Builder.Dialogs.Rules/Steps/OnCatch.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs;
+
+namespace Microsoft.Bot.Builder.Dialogs.Rules.Steps
+{
+    public class OnCatchConditional
+    {
+        public string EventName { get; set; }
+        public List<IDialog> Steps { get; set; }
+    }
+
+    public class OnCatch : DialogCommand, IDialogDependencies
```

Comment:
Why is this one OK to be called OnXXXX ?  WTF?
CatchEvent perhaps?  Why not a rule?  I am confused?